flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableSource.java

```diff
@@ -247,6 +255,36 @@ public void applyPartitions(List<Map<String, String>> remainingPartitions) {
         }
     }
 
+    @Override
+    public List<String> applyDynamicFiltering(List<String> candidateFilterFields) {
+        if (catalogTable.getPartitionKeys() != null
+                && catalogTable.getPartitionKeys().size() != 0) {
+            checkArgument(
+                    !candidateFilterFields.isEmpty(),
+                    "At least one field should be provided for dynamic filtering");
+            checkState(
+                    dynamicFilterPartitionKeys == null,
+                    "Dynamic filtering should not be applied twice.");
+
+            // only accept partition fields to do dynamic partition pruning
+            this.dynamicFilterPartitionKeys = new ArrayList<>();
+            for (String field : candidateFilterFields) {
+                if (catalogTable.getPartitionKeys().contains(field)) {
+                    dynamicFilterPartitionKeys.add(field);
```

Comment:
We should ignore the fields which are unsupported type, e.g. TIME_WITHOUT_TIME_ZONE, ARRAY, etc